### ISSUE TITLE
Javascript API in flowplayer 

### ISSUE THREAD

**Issue by joseluismg** [mangui/flashls#403]
Hi,

sorry for the newbie question, I am new in using this plugin. Is there any Javascript API to use with flowplayer?  More specifically I am trying to get a callback when there is a switch in the stream being played. What I mean is if  there are calls to onStreamSwitch or any similar and if not if this is going to be in the added in the future?

many thanks!

**Comment by mangui** [comment]
Hi, it might be feasible, maybe @blacktrash could answer ?

**Comment by phloxic** [comment]
I assume the `onMetaData`, and `onMetaDataChange` clip events - http://flash.flowplayer.org/documentation/events/clip.html - would be the places to send `clip.metaData` http://flash.flowplayer.org/documentation/api/clip.html#properties

The events are fired, but flashls provided clips always have `metaData` undefined. - Aside: I wonder whether therefore _setting_ `clip.metaData` to `false` to tell the player not to expect meta data would currently help in some problematic usage scenarios

Here is how this works (or should work) for the bitrate select plugin: http://flowplayer.blacktrash.org/test/brmenu.html

**Comment by joseluismg** [comment]
yes, exactly. Would it be possible to have a brmenu like this working with the flashls plugin?

Because I have seen this issue, but It seems open: https://github.com/mangui/flashls/issues/124

**Comment by phloxic** [comment]
@joseluismg - But the example deals with explicit URLs for manual selection. I have no idea whether and how flashls' API could be mapped to that.
If this can't be done from flashls' side I doubt that it will be implemented from the core player side in Flowplayer Flash - unless you do it yourself ;-)
We are _planning_ to implement this for Flowplayer HTML5, espcially with hls.js (problem: doesn't work in Safari yet). But it will probably take a while.

**Comment by joseluismg** [comment]
@mangui do you think integrating the quality level selection is something feasible or something that will be included at some point? or as alternative do you plan to inlcude a javascript API to handle the quality externally
.

**Comment by mangui** [comment]
that might be feasible, but you will have to do it yourself.
you can inspire from https://github.com/flowplayer/flash/tree/master/plugins/bitrateselect/src/actionscript/org/flowplayer/bitrateselect, which listen to custom bitrates property on Clip : 
https://github.com/flowplayer/flash/blob/master/plugins/bitrateselect/src/actionscript/org/flowplayer/bitrateselect/BitrateSelectProvider.as#L279

there are commented codes available here to dispatch quality level to FlowPlayer : https://github.com/mangui/flashls/blob/dev/src/org/mangui/flowplayer/HLSStreamProvider.as#L125-L141

and then you would have to implement https://github.com/mangui/flashls/blob/dev/src/org/mangui/flowplayer/HLSStreamProvider.as#L525-L530

**Comment by joseluismg** [comment]
@mangui can you pm me, please? joseluismg@gmail.com

**Comment by joseluismg** [comment]
do any of you know any developer available to work in this feature? The company where I work is happy to sponsor this feature.
